giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/aspects/access/RestSecurityAspect.java

```diff
@@ -287,8 +288,40 @@ public Object checkImageAccess(ProceedingJoinPoint joinPoint, ImageAccessCheck c
         
         return joinPoint.proceed(arguments);
     }
-    
-    
+
+    @Around("within(edu.asu.diging.gilesecosystem.web.rest..*) && @annotation(check)")
+    public Object checkDocumentOwner(ProceedingJoinPoint joinPoint, DocumentOwnerCheck check) throws Throwable {
+
+        UserTokenObject userTokenObj = extractUserTokenInfo(joinPoint, check.github(), check.value());
+
+        User user = userTokenObj.user;
+        String token = userTokenObj.token;
+        String docId = userTokenObj.parameter;
+
+        if (user == null) {
+            throw new AspectMisconfigurationException("User object is missing in method.");
+        }
+
+        IDocument doc = filesManager.getDocument(docId);
+        if (doc == null) {
+            return new ResponseEntity<String>(HttpStatus.NOT_FOUND);
+        }
```

Comment:
removed doc null check from controller. With token check, this method also check for document owner. User requesting document access change should be owner of the document irrespective of current access type. We have previous DocumentAccessCheck annotation but it does not check document owner for public documents.